src/crate/crash/repl.py

```diff
@@ -179,14 +180,32 @@ def __init__(self, cmd):
     def __call__(self, buffer):
         if not self.cmd.should_autocapitalize():
             return
-        last_word = buffer.document.get_word_before_cursor(WORD=True)
-        if not last_word.isupper() and last_word.lower() in SQLCompleter.keywords:
-            buffer.delete_before_cursor(len(last_word))
-            buffer.insert_text(last_word.upper(), fire_event=False)
-            self.last_changed = last_word
-        elif self.last_changed and last_word.startswith(self.last_changed.upper()):
-            buffer.delete_before_cursor(len(last_word))
-            buffer.insert_text(self.last_changed + last_word[-1:], fire_event=False)
+
+        def keyword_replacer(match):
+            if match.group(0).lower() in SQLCompleter.keywords:
+                return match.group(0).upper()
+            else:
+                return match.group(0)
+
+        def is_prefix(a, b):
```

Comment:
this method can be on module level, it does not require the context of the method